docs/src/components/Sidebar.tsx

```diff
@@ -147,19 +165,32 @@ export interface Props {
 
 const Sidebar: React.FC<Props> = props => {
   const data: QueryResult = useStaticQuery(query);
+
+  const [isSearching, setIsSearching] = React.useState(false);
+  const [filter, setFilter] = React.useState<string[] | null>(null);
+
   const currentPath = props.location.pathname;
-  const sections = getSections(data);
+  const sections = getSections(data, filter);
 
   return (
     <StyledSidebar className="sidebar" isOpen={props.isOpen}>
+      <Search
+        onSearchResults={results => {
+          setFilter(results);
+          setIsSearching(results !== null);
+        }}
+      />
       {sections.map(section => (
         <Section
           key={section.slug}
           section={section}
+          searching={isSearching}
           active={currentPath.startsWith(section.slug)}
           currentPath={currentPath}
         />
       ))}
+
+      {filter != null && filter.length === 0 && <p>{"No Results :("}</p>}
```

Comment:
I'm slightly confused by the variable name filter here. Does it mean filtered results, or terms to filter out? 
Should this line be a check if results/sections are empty?

Reply:
`filter` is either `null` or a list of document ids that should be displayed. It is basically the results. I'll rename the variable.